src/hooks/market/useMarkets.ts

```diff
@@ -2,18 +2,20 @@ import { useContractRead, useContractReads } from "wagmi";
 import registryContractJson from "../../abi/Registry.json";
 import { BigNumber, ethers } from "ethers";
 import { useEffect, useState } from "react";
+import { useConfig } from "../../providers/Config";
 import api from "../../apis/Api";
 
-const registryContractAddress = process.env.VITE_REGISTRY_CONTRACT;
-if (!registryContractAddress)
-  throw new Error("No VITE_REGISTRY_CONTRACT provided");
+const useMarketAddressesFromContract = () => {
+  const config = useConfig();
+  const registryContractAddress = config?.addresses?.registry;
+  if (!registryContractAddress)
+    throw new Error("No REGISTRY_CONTRACT provided");
```

Comment:
we dont want to throw if there is no address provided -- `undefined` happens if the config is loading, i think you can just pass an undefined address into the `registryContract` object?

Reply:
alternatively, you could pass `config` in as a prop and load it in the parent component